File: ruby/Asset/Snippets/UpdateSnippetContent.rb
require 'rest_client'
require 'json'

host = "CHANGE ME"
client_id = "CHANGE ME"
client_secret = "CHANGE ME"

def get_token(host, client_id, client_secret)
  url = "#{host}/identity/oauth/token?grant_type=client_credentials&client_id=#{client_id}&client_secret=#{client_secret}"
  response = RestClient.get url
  json = JSON.parse(response)
  return json["access_token"]
end

id = 10 #id of snippet, required
params = {
  :access_token => get_token(host, client_id, client_secret),
  :type => "HTML",
  :content => "<div>This is HTML content</div>"
}

response = RestClient.post "#{host}/rest/asset/v1/snippet/#{id}/content.json", params

puts response

File: php/Asset/Snippets/UpdateSnippetContent.php
<?php
/*
   UpdateSnippetContent.php

   Marketo REST API Sample Code
   Copyright (C) 2016 Marketo, Inc.

   This software may be modified and distributed under the terms
   of the MIT license.  See the LICENSE file for details.
*/
$snippet = new UpdatesnippetContent();
$snippet->id = 9;
$snippet->type = "Text";
$snippet->content = "This is Content";
print_r($snippet->postData());

class UpdatesnippetContent{
	private $host = "CHANGE ME";
	private $clientId = "CHANGE ME";
	private $clientSecret = "CHANGE ME";
	public $id;//id of the snippet to update, required
	//see docs for explanation
	public $type;//type of content, Text, HTML, or DynamicContent
	public $content;//content to input
	
	public function postData(){
		$url = $this->host . "/rest/asset/v1/snippet/" . $this->id . "/content.json?access_token=" . $this->getToken();
		$ch = curl_init($url);
		$requestBody = $this->bodyBuilder();
		curl_setopt($ch,  CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('accept: application/json'));
		curl_setopt($ch, CURLOPT_POST, 1);
		curl_setopt($ch, CURLOPT_POSTFIELDS, $requestBody);
		curl_getinfo($ch);
		$response = curl_exec($ch);
		return $response;
	}
	
	private function getToken(){
		$ch = curl_init($this->host . "/identity/oauth/token?grant_type=client_credentials&client_id=" . $this->clientId . "&client_secret=" . $this->clientSecret);
		curl_setopt($ch,  CURLOPT_RETURNTRANSFER, 1);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array('accept: application/json'));
		$response = json_decode(curl_exec($ch));
		curl_close($ch);
		$token = $response->access_token;
		return $token;
	}
	private function bodyBuilder(){
		$requestBody = "&type=" . $this->type . "&content=" . $this->content;
		return $requestBody;
	}
}